default-cluster-templates/baseline-k3s.json

```diff
@@ -16,65 +16,65 @@
-            "make -p /var/lib/kubelet/static-pods",
-            "sed -i '/kubelet-arg:/a\\- pod-manifest-path=/var/lib/kubelet/static-pods' /etc/rancher/k3s/config.yaml"
-          ],
-          "postK3sCommands": [
-            "kubectl apply -f https://raw.githubusercontent.com/projectcalico/calico/v3.29.3/manifests/calico.yaml"
-          ],
-          "privateRegistriesConfig": {},
-          "serverConfig": {
-            "cni": "calico",
-            "cniMultusEnable": true,
-            "disableComponents": {
-              "kubernetesComponents": [
-                "cloudController"
-              ]
-            },
-            "kubeApiServer": {
-              "extraArgs": [
-                "--feature-gates=PortForwardWebsockets=true",
-                "--tls-cipher-suites=TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384"
-              ]
-            },
-            "kubeControllerManager": {
-              "extraArgs": null
-            },
-            "kubeScheduler": {
-              "extraArgs": null
-            },
-            "etcd": {
-              "exposeMetrics": false,
-              "backupConfig": {
-                "retention": "5",
-                "scheduleCron": "0 */5 * * *"
-              },
-              "customConfig": {
+          "kthreesConfigSpec": {
+            "files": [
+              {}
+            ],
+            "agentConfig": {
+              "kubelet": {
                 "extraArgs": [
-                  "cipher-suites=[TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384]"
+                  "--topology-manager-policy=best-effort",
+                  "--cpu-manager-policy=static",
+                  "--reserved-cpus=1",
+                  "--max-pods=250",
+                  "--tls-cipher-suites=TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384"
                 ]
               }
-            }
+            },
+            "preK3sCommands": [
+              "mkdir -p /etc/systemd/system/k3s-server.service.d",
+              "echo '[Service]\nEnvironmentFile=/etc/environment' > /etc/systemd/system/k3s-server.service.d/override.conf",
+              "mkdir -p /var/lib/kubelet/static-pods",
+              "sed -i '/kubelet-arg:/a\\- pod-manifest-path=/var/lib/kubelet/static-pods' /etc/rancher/k3s/config.yaml",
+              "sudo cp /var/lib/rancher/k3s/agent/pod-manifests/connect-agent.yaml /var/lib/kubelet/static-pods/connect-agent.yaml"
+            ],
+            "postK3sCommands": [
+              "kubectl apply -f https://raw.githubusercontent.com/projectcalico/calico/v3.29.3/manifests/calico.yaml",
+              "sudo mount --make-shared /sys && sudo mount --make-shared / && sudo mount --make-shared /run"
```

Comment:
question: what is this mount needed for?

Reply:
It seems that happen when we deploy calico. One pods get `CreateContainerError` and the following warning appeared in pod describe:
```
 Warning  Failed     5m11s                 kubelet            Error: failed to generate container "b261b5bd37f652cb6a52f7c2e12335007ccb506303dbb9cd4105095ba9e2f1e7" spec: failed to generate spec: path "/sys/fs/" is mounted on "/sys" but it is not a shared mount
 ```
so,  /sys is not a shared mount, and kubelet cannot propagate. I am still not sure if this is related any apparmor, but what I can say, this issue does not appear when k3s cluster is created in other environment, for instance on WSL, and with calico deployment.